src/EditorFeatures/Core/ExternalAccess/VSTypeScript/Api/VSTypeScriptDebugLocationInfo.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
+{
+    internal struct VSTypeScriptDebugLocationInfo
+    {
+        public readonly string Name;
+        public readonly int LineOffset;
+
+        public VSTypeScriptDebugLocationInfo(string name, int lineOffset)
+        {
+            Debug.Assert(name != null);
+            Name = name;
+            LineOffset = lineOffset;
+        }
+
+        internal bool IsDefault => Name == null;
```

Comment:
This should be updated to use the wrapper pattern.

https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingInvocationReasonsWrapper.cs

If the type doesn't need to be _constructed_ by the partner then it only needs the one constructor.

Reply:
I just copied F# implementation of the same:
https://github.com/dotnet/roslyn/blob/af9d30a89b5aeed375f1c4f8f459eb7ac7a74359/src/Tools/ExternalAccess/FSharp/Editor/Implementation/Debugging/IFSharpLanguageDebugInfoService.cs#L10

Reply:
The type needs to be constructed. 

I'm not seeing the benefits of the wrapper pattern. Doesn't `UnderlyingObject` expose the type that they shouldn't access?

Reply:
> I'm not seeing the benefits of the wrapper pattern. 

Helps link related code together, and minimize duplication.

> Doesn't UnderlyingObject expose the type that they shouldn't access?

No. The analyzer will not allow the property to be used in partner code so it's effectively a proper `internal` visibility.